pydantic/fields.py

```diff
@@ -280,6 +301,34 @@ def _validate_json(self, v: str, loc: Tuple[str, ...]) -> Tuple[Optional[Any], O
         except (ValueError, TypeError) as exc:
             return v, ErrorWrapper(exc, loc=loc, config=self.model_config)
 
+    def _validate_sequence(
```

Comment:
I think you can probably achieve this by modifying `_validate_list_set` (perhaps it could be renamed to `_validate_sequence`.

Like List, Set and Tuple, it should be able to validate list, set, tuple and generators.

Reply:
I've fixed that